Example/MoyaComparison/MoyaComparison/Utils/BarButtonItemFactory.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2017 niji
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import UIKit
+
+final class BarButtonItemFactory: UIBarButtonItem {
```

Comment:
A "Factory" isn't a "Button". That's like if you said that a "Car Factory is a Car"…

If it were really a "Factory" that would be a class inheriting from nothing particular with functions returning `UIBarButtonItem` instances.

But here what this class does is it's a special kind of `UIBarButtonItem` that handles its action using a block, so better change the name to something like `BlockBarButtonItem`.